mypy/report.py

```diff
@@ -156,14 +160,66 @@ def on_file(self,
         visitor = stats.StatisticsVisitor(inferred=True, filename=tree.fullname(),
                                           typemap=type_map, all_nodes=True)
         tree.accept(visitor)
+        self.any_types_counter[tree.fullname()] = visitor.type_of_any_counter
         num_unanalyzed_lines = list(visitor.line_map.values()).count(stats.TYPE_UNANALYZED)
         # count each line of dead code as one expression of type "Any"
-        num_any = visitor.num_any + num_unanalyzed_lines
-        num_total = visitor.num_imprecise + visitor.num_precise + num_any
+        num_any = visitor.num_any_exprs + num_unanalyzed_lines
+        num_total = visitor.num_imprecise_exprs + visitor.num_precise_exprs + num_any
         if num_total > 0:
             self.counts[tree.fullname()] = (num_any, num_total)
 
     def on_finish(self) -> None:
+        self._report_any_exprs()
+        self._report_types_of_anys()
+
+    def _report_types_of_anys(self) -> None:
+        # On the following line, mypy complains about implicit generic Any and wants us to use
+        # typing.Counter() instead of collections.Counter().
+        # We cannot use typing.Counter() because it doesn't work with python 3.5 and older.
+        total_counter = collections.Counter()  # type: ignore
+        for counter in self.any_types_counter.values():
+            for any_type, value in counter.items():
+                total_counter[any_type] += value
+        any_types_column_names = collections.OrderedDict([
+            (TypeOfAny.implicit, "Unannotated"),
+            (TypeOfAny.explicit, "Explicit"),
+            (TypeOfAny.from_unimported_type, "Unimported"),
+            (TypeOfAny.from_omitted_generics, "Omitted Generics"),
+            (TypeOfAny.from_error, "Error"),
+            (TypeOfAny.special_form, "Special Form"),
+            (TypeOfAny.from_another_any, "From Another Any"),
+        ])  # type: collections.OrderedDict[TypeOfAny, str]
+        file_column_name = "Name"
```

Comment:
I wonder if the rest of this function could be abstracted out/combined with similar looking code from the function below.